src/main/java/seedu/address/logic/commands/ExportCommand.java

```diff
@@ -63,24 +68,84 @@ public CommandType getCommandType() {
+    }
+
     @Override
     public CommandResult execute(Model model) throws CommandException {
         List<Student> studentList = model.getFilteredStudentList();
 
+        Path dataFilePath = baseDir.resolve(filename + ".csv");
+        Path homeFilePath = getHomeFilePath(filename);
+
         // Create directories if they don't exist
         try {
             Files.createDirectories(baseDir);
         } catch (IOException e) {
             throw new CommandException(String.format(MESSAGE_FAILURE, "Could not create directory: " + e.getMessage()));
         }
 
-        Path filePath = baseDir.resolve(filename + ".csv");
+        // Check both locations for existing files when force flag is not set
+        if (!isForceExport) {
+            if (Files.exists(dataFilePath)) {
+                throw new CommandException(String.format(MESSAGE_FILE_EXISTS, dataFilePath));
+            }
+            if (Files.exists(homeFilePath)) {
+                throw new CommandException(String.format(MESSAGE_HOME_FILE_EXISTS, homeFilePath));
+            }
+        }
+
+        try {
+            // Write to data directory
+            writeCsvFile(dataFilePath, studentList);
+
+            // Copy to home directory
+            try {
+                if (isForceExport) {
+                    Files.copy(dataFilePath, homeFilePath, StandardCopyOption.REPLACE_EXISTING);
+                } else {
+                    Files.copy(dataFilePath, homeFilePath);
+                }
+                return new CommandResult(String.format(MESSAGE_SUCCESS_WITH_COPY,
+                        studentList.size(), dataFilePath, homeFilePath),
+                        COMMAND_TYPE);
+            } catch (FileAlreadyExistsException e) {
+                // If file exists in home directory and force flag is not set
+                // Delete the data directory file and throw exception
+                Files.deleteIfExists(dataFilePath);
+                throw new CommandException(String.format(MESSAGE_HOME_FILE_EXISTS, homeFilePath));
+            } catch (IOException e) {
+                // If home directory copy fails for other reasons, return success with data file only
+                return new CommandResult(String.format(MESSAGE_SUCCESS, studentList.size(), dataFilePath),
+                        COMMAND_TYPE);
+            }
+        } catch (IOException e) {
+            // Clean up any partially created files
+            try {
+                Files.deleteIfExists(dataFilePath);
+            } catch (IOException ignored) {
+                // Ignore cleanup errors
+            }
+            throw new CommandException(String.format(MESSAGE_FAILURE, e.getMessage()));
+        }
+    }
 
-        // Check if file exists and force flag is not set
-        if (!isForceExport && Files.exists(filePath)) {
-            throw new CommandException(String.format(MESSAGE_FILE_EXISTS, filePath));
+    String escapeSpecialCharacters(String data) {
```

Comment:
Out of curiosity, what does this do? Why do we need to escape special characters?

Reply:
It's more of a utility function to help with .csv formatting throughout the file.